bpfd/src/bpf.rs

```diff
@@ -961,21 +981,22 @@ impl BpfManager {
                 map.used_by.swap_remove(index);
             }
 
-            // If map owner program still exists update it's maps_used_by_field
-            if let Some(program) = self.programs.get_mut(&index) {
-                // TODO(astoycos) remove external self.maps, keep all map tracking info in Program
-                // update programs map_used_by
-                program
-                    .data_mut()?
-                    .set_maps_used_by(Some(map.used_by.clone()));
-            }
-
             if map.used_by.is_empty() {
+                // No more programs using this map, so remove the entry from the map list.
                 let path = calc_map_pin_path(index);
                 self.maps.remove(&index.clone());
                 fs::remove_dir_all(path)
                     .await
                     .map_err(|e| BpfdError::Error(format!("can't delete map dir: {e}")))?;
+            } else {
+                // Update all the programs still using the same map with the updated map_used_by.
+                for id in map.used_by.iter() {
+                    if let Some(program) = self.programs.get_mut(id) {
+                        if let Ok(data) = program.data_mut() {
+                            data.set_maps_used_by(Some(map.used_by.clone()));
+                        }
```

Comment:
Change into a on-liner as above

Reply:
clippy and fmt forced it back to what I had originally.